app/proguard-rules.pro

```diff
@@ -9,6 +9,10 @@
 -keep class com.sun.jna.** { *; }
 -keep class * implements com.sun.jna.** { *; }
 
+# Prevent the sqlcipher library from being obfuscated/optimised when building in
+# release mode, as it causes the library to raise exceptions.
+-keep class net.sqlcipher.** { *; }
```

Comment:
maybe add this to the module who references the `sqlcipher` library.
We did not play yet with Proguard on this project, so I am not 100% sure it will work as expected.

Reply:
It looks like `encrypted-db` and `session-storage` are the libraries that import `sqlcipher`. Are you suggesting I add a `proguard-rules.pro` file for each of those libraries? https://developer.android.com/studio/build/shrink-code

Reply:
> We did not play yet with Proguard on this project, so I am not 100% sure it will work as expected.

I can confirm that it does work on a local branch. 

@anoadragon453 can you apply [this patch](https://github.com/vector-im/element-x-android/files/10897510/0001-Move-SQLCipher-proguard-rules-to-consumer-rules-so-t.patch) to achieve what Benoit mentioned above? I have a working branch where these changes work, but since you already opened the PR it feels wrong to close it just to replace it with another one from us.

Also @bmarty any idea why the CI is failing with 'unauthorized' errors for him? Is it just because the commit message is missing a sign-off or do we have something preventing the CI from running for external users?

Reply:
> @anoadragon453 can you apply [this patch](https://github.com/vector-im/element-x-android/files/10897510/0001-Move-SQLCipher-proguard-rules-to-consumer-rules-so-t.patch) to achieve what Benoit mentioned above? I have a working branch where these changes work, but since you already opened the PR it feels wrong to close it just to replace it with another one from us.

That patch appears to work, thanks!

And generally I'd advise not feeling afraid to close contributor PRs and supersede them with your own. We take this approach on matrix-org/synapse and contributors are generally happy that their bug was fixed/feature was implemented. It helps to thank them in the changelog still though :)

Reply:
> Should I close the PR then and create a new one?

Yes, please, but maybe merge https://github.com/vector-im/element-x-android/pull/136 first to avoid conflict.